subprojects/base-services/src/main/java/org/gradle/internal/resources/DefaultResourceLockCoordinationService.java

```diff
@@ -82,29 +83,62 @@ public ResourceLockState getCurrent() {
         }
     }
 
-    @Override
-    public void notifyStateChange() {
-        synchronized (lock) {
-            lock.notifyAll();
+    private void maybeNotifyStateChange(DefaultResourceLockState resourceLockState) {
+        if (resourceLockState.getUnlockedResources() != null && !resourceLockState.getUnlockedResources().isEmpty()) {
```

Comment:
⭕️ if we add `hasUnLockedResources()`, we don't need to handle null/isEmpty explicitly here